src/pages/permissions/panels/EditGroupIdentitiesPanel.tsx

```diff
@@ -0,0 +1,341 @@
+        modifiedIdentities.add(identity);
+      }
+    }
+
+    for (const identity of identityIdsInSomeGroups) {
+      if (!indeterminateIdentities.has(identity)) {
+        modifiedIdentities.add(identity);
+      }
+    }
+
+    for (const identity of identityIdsInNoGroups) {
+      if (selectedIdentities.has(identity)) {
+        modifiedIdentities.add(identity);
+      }
+    }
+
+    return modifiedIdentities;
+  };
+
+  const modifyIdentities = (
+    newIdentities: string[],
+    isUnselectAll?: boolean,
+  ) => {
+    if (isUnselectAll) {
+      saveToPanelHistory({
+        identitiesAdded: new Set(),
+        identitiesRemoved: new Set(
+          nonTlsIdentities.map((identity) => identity.id),
+        ),
+      });
+    } else {
+      saveToPanelHistory({
+        identitiesAdded: new Set(newIdentities),
+        identitiesRemoved: new Set(),
+      });
+    }
+  };
+
+  const toggleRow = (rowName: string) => {
+    const isRowSelected = selectedIdentities.has(rowName);
+    const isRowIndeterminate = indeterminateIdentities.has(rowName);
+
+    const identitiesAdded = new Set(desiredState.identitiesAdded);
+    const identitiesRemoved = new Set(desiredState.identitiesRemoved);
+
+    if (isRowSelected || isRowIndeterminate) {
+      identitiesAdded.delete(rowName);
+      identitiesRemoved.add(rowName);
+    } else {
+      identitiesAdded.add(rowName);
+      identitiesRemoved.delete(rowName);
+    }
+
+    saveToPanelHistory({
+      identitiesAdded,
+      identitiesRemoved,
+    });
+  };
+
+  const closePanel = () => {
+    panelParams.clear();
+    notify.clear();
+    setConfirming(false);
+  };
+
+  const closeModal = () => {
+    notify.clear();
+    setConfirming(false);
+  };
+
+  const modifiedIdentities = calculatedModifiedIdentities();
+
+  const headers = [
+    { content: "Identity", sortKey: "name", role: "rowheader" },
+    {
+      content: "",
+      role: "rowheader",
+      "aria-label": "Modified status",
+      className: "modified-status",
+    },
+  ];
+
+  const filters: PermissionIdentitiesFilterType = {
+    queries: searchParams.getAll(QUERY),
+    authMethod: searchParams.getAll(AUTH_METHOD),
+  };
+
+  const filteredIdentities = nonTlsIdentities.filter((identity) => {
+    if (
+      !filters.queries.every(
+        (q) =>
+          identity.name.toLowerCase().includes(q) ||
+          identity.id.toLowerCase().includes(q),
+      )
+    ) {
+      return false;
+    }
+
+    if (
+      filters.authMethod.length > 0 &&
+      !filters.authMethod.includes(identity.authentication_method)
+    ) {
+      return false;
+    }
+
+    return true;
+  });
+
+  const rows = filteredIdentities.map((identity) => {
+    const modifiedTitle = desiredState.identitiesAdded.has(identity.id)
+      ? "Group will be added to all selected identities"
+      : desiredState.identitiesRemoved.has(identity.id)
+        ? "Group will be removed from all selected identities"
```

Comment:
Copy and paste error? We are selecting identities for groups. Also need to consider the singular/plural case.

Reply:
yeah that was a copy mistake.